pkg/server/application_api/application_api_test: TestStatusAPIContentionEvents failed

pkg/server/application_api/application_api_test.TestStatusAPIContentionEvents [failed](https://mesolite.cluster.engflow.com/invocations/default/662533a4-7b2e-4c8e-a7bc-9069049c67c7?testReportRun=3&testReportShard=5&testReportAttempt=1#targets-Ly9wa2cvc2VydmVyL2FwcGxpY2F0aW9uX2FwaTphcHBsaWNhdGlvbl9hcGlfdGVzdA==) on release-25.2 @ [72b53a17b9ca2eb0b58c00364f43442e0b150731](https://github.com/cockroachdb/cockroach/commits/72b53a17b9ca2eb0b58c00364f43442e0b150731):


```
=== RUN   TestStatusAPIContentionEvents
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestStatusAPIContentionEvents1572310294
    test_log_scope.go:76: use -show-logs to present logs inline
    test_server_shim.go:154: automatically injected a shared process virtual cluster under test; see comment at top of test_server_shim.go for details.
    contention_test.go:121: query '
          SELECT count(*)
          FROM crdb_internal.transaction_statistics
          WHERE
            (statistics -> 'execution_statistics' -> 'contentionTime' ->> 'mean')::FLOAT > 0
            AND app_name = 'contentionTest'
        ': expected:
        1
        
        got:
        0
        
    panic.go:635: -- test log scope end --
test logs left over in: outputs.zip/logTestStatusAPIContentionEvents1572310294
--- FAIL: TestStatusAPIContentionEvents (48.56s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=3</code>
 - <code>shard=5</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/obs-prs
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestStatusAPIContentionEvents.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-60535

--- a/pkg/server/application_api/contention_test.go
+++ b/pkg/server/application_api/contention_test.go
@@ -129,7 +129,7 @@ SET TRACING=off;
 	require.True(t, found,
 		"expect to find contention event for table %d, but found %+v", testTableID, resp)
 
-	server1Conn.CheckQueryResults(t, `
+	server1Conn.CheckQueryResultsRetry(t, `
   SELECT count(*)
   FROM crdb_internal.statement_statistics
   WHERE
@@ -137,7 +137,7 @@ SET TRACING=off;
     AND app_name = 'contentionTest'
 `, [][]string{{"1"}})
 
-	server1Conn.CheckQueryResults(t, `
+	server1Conn.CheckQueryResultsRetry(t, `
   SELECT count(*)
   FROM crdb_internal.transaction_statistics
   WHERE

